### ISSUE TITLE
"Merge attributes from..." loses attributes tags

### ISSUE THREAD

**Issue by imereau** [MISP/MISP#2682]
When merging into one event#1 the attributes of another event#2, it seems that the tagging of the attributes from event#2 are lost during the operation, which is very annoying if you have a large list of attributes with multiple tags..

Also, I would expect that if all attributes of an event#2 are merged into an event#1, the event#2 is deleted to avoid duplicates.

### Work environment

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)   |    ubuntu
| OS version (client)      | 10
| PHP version               | 7.0
| MISP version / git hash   | 2.4.82, f19a3a7c1b65f9c1b01f29017166d0517e338512
| Browser                   | FF 47.0.1

### Expected behavior
tagging information of attributes is persisted during the merge operation

### Actual behavior
tagging information lost

### Steps to reproduce the behavior

- create an event#1
- add any attribute and mark it with a tag, say TLP:Green
- create an event#2
- add any attribute and mark it with a tag, say TLP:Red
- go into event#1
- On the left hand pane, click "Merge attributes from..." and input the event ID corresponding to event#2
- Click view event#1
- attribute of event#2 has not tags anymore

**Comment by RichieB2B** [comment]
I agree that the attribute tags should be copied. The deletion of the original event however should definitely be optional (if implemented at all). Our use case for merging events is when the original event is owned by a different (unresponsive) organization and we want to extend/fix the event.

**Comment by imereau** [comment]
OK understood for the deletion of events. Regarding the persistence of the tags, any idea by when that's gonna be implemented (if ever). I'm busy with some migration, just to know what I can expect. 

Thanks a lot in advance!
Irving

**Comment by RichieB2B** [comment]
We don’t use attribute tags much, so I have no plans to work on this soon.

**Comment by imereau** [comment]
ok, no worries, I can work around that with the API.
One last thing.. I also noticed that objects are not kept as such when merging, they are split into their attributes.. Is it anyhow possible to merge also objects from an event into another one? That would be very handy! :-)

Thanks!

**Comment by certbe-trey** [comment]
Why isn't there an option to merge objects, a la "merge attributes from..."? And furthermore, when merging attributes and/or objects, it would be **gorgeous** if there were a checkbox option to also include tags and/or galaxies in the merge.

**Comment by iglocska** [comment]
Generally not a big fan of the merge attributes from feature. Sadly it promotes a pretty nasty practice of misappropriating data from other producers that we've seen in some cases so it's kind of low prio for us.

**Comment by certbe-trey** [comment]
Ah, but say for example you're a national CERT working a big investigation and you realize at some point that you need to decompose a big event into smaller ones (or merge smaller ones together) - I get what you're saying about not wanting to promote plagiarizing others' data, but from an internal workflow perspective there's a whole other angle.

**Comment by iglocska** [comment]
We could do something specifically for events created by the same organisation to avoid the stealing part - but it's not exactly a trivial amount of work and there's a bunch of higher prio stuff on our plate atm. Generally we try to prioritise stuff that comes up more frequently from different users (or stuff that someone pays us for to fast-track ofc...) - so for now it will have to be thrown on the pile of nice-to-haves, unless someone jumps in and does a PR. :(
